### ISSUE TITLE
no-shadowed-variable: request for option to whitelist some shadowed variables

### ISSUE THREAD

**Issue by benny-medflyt** [palantir/tslint#3049]
Consider callback-heavy code like the following:

```typescript
import * as fs from "fs";

function main() {
    fs.readFile("file1", (err, file1) => {
        if (err) {
            console.error(err);
            process.exit();
        }

        fs.readFile("file2", (err, file2) => {  // "err" is shadowed variable!
            if (err) {
                console.error(err);
                process.exit();
            }

            console.log(file1, file2);
        });
    });
}
```

This triggers: `no-shadowed-variable` `Shadowed variable: 'err'`.

I would like an option for `no-shaded-variable` that would allow to whitelist variables if they are of a specified type. I'm not sure how doable this is since `no-shadowed-variable` is not "Requires Type Info" enabled, but I have a use-case where this would be convenient.

**Comment by ajafff** [comment]
I also thought about that. I don't think this rule will be enabled to use the type checker just to ignore variables of certain types.

We could add a (global) whitelist. The rule would ignore that name no matter what kind of declaration it is.

Or it could be added on top of #3030. For example that would let you whitelist `err` as parameter name, but still show an error for local variables of the same name.

What do you prefer?

**Comment by doberkofler** [comment]
I would prefer a global whitelist

**Comment by JoshuaKGoldberg** [comment]
@doberkofler an 'ignore' option was added in #3030, does that fix it for you?

**Comment by JoshuaKGoldberg** [comment]
Awesome! Marking this issue as closed since that sounds like a sufficient workaround. Please do post here if that's not the case and we can take another look.
